docs/networking.rst

```diff
@@ -240,3 +240,60 @@ having to pass the path every time you send data to the peer::
             await unix_dg.send(b'Hi there!\n')
 
     run(main)
+
+Abstracting remote connections using Connectables
+-------------------------------------------------
+
+AnyIO offers a hierarchy of classes implementing either the
+:class:`ObjectStreamConnectable` or :class:`ByteStreamConnectable` interfaces which
+lets developers abstract out the connection mechanism for network clients.
+For example, you could create a network client class like this::
+
+    from os import PathLike
+    from ssl import SSLContext
+
+    from anyio.abc import ByteStreamConnectable, as_connectable
+
+
+    class MyNetworkClient:
+        def __init__(
+            self,
+            connectable: ByteStreamConnectable | tuple[str, int] | str | PathLike[str],
```

Comment:
do we want to provide a TypeAlias for this?

Reply:
Perhaps as a follow-up. What would I call it?